### PR TITLE
[MB-13824] Create an ADR for Move History Templates Event Name

### PR DESCRIPTION
**Pull request by afoon**
https://dp3.atlassian.net/browse/MB-13825

### CODE REVIEW

File: docs/adrs/0073-move-history-details-refactor.md
Comment (kctruss) [inline docs/adrs/0073-move-history-details-refactor.md:2]: **TODO:** Our ADR work overlapped, so this number should update to `0074`

File: docs/adrs/0073-move-history-details-refactor.md
Comment (kctruss) [inline docs/adrs/0073-move-history-details-refactor.md:5]: **typo:** "templated" -> "templates"

File: docs/adrs/0073-move-history-details-refactor.md
Comment (kctruss) [inline docs/adrs/0073-move-history-details-refactor.md:52]: **typo:** "matched" -> "match"

File: docs/adrs/0073-move-history-details-refactor.md
Comment (kctruss) [inline docs/adrs/0073-move-history-details-refactor.md:19]: **typo:** "resused" -> "reused"

File: docs/adrs/0073-move-history-details-refactor.md
Comment (kctruss) [inline docs/adrs/0073-move-history-details-refactor.md:104]: **typos:** 

- "consildate" -> "consolidate" 
- "labelled" -> "labeled"
- "muliple" -> "multiple"
- "strusture" -> "structure"
- "indivcidual" -> "individual"

**Review by kctruss — CHANGES_REQUESTED** [review]
This is really great, @afoon ! Just spotted a few typos, and realized I should have checked in with you earlier about our ADR # potential collision, so that one is on me.

**Review by kctruss — APPROVED** [review]
🎉 lgtm!

File: docs/adrs/0074-move-history-details-refactor.md
Comment (ronaktruss) [inline docs/adrs/0074-move-history-details-refactor.md:13]: (Thought/suggestion): It could be worthwhile to include a link to the version of the file at the time of writing this file as well. Maybe something like along the lines of
```suggestion
To be able to handle changed values. We will first need to refactor the existing Template Manager[[current](https://github.com/transcom/mymove/blob/master/src/constants/MoveHistory/TemplateManager.js) | [time of writing](https://github.com/transcom/mymove/blob/bc0369eec482fca3d132a9702712312bb224f04d/src/constants/MoveHistory/TemplateManager.js)]. We will need the template manager to be able to three things:
```

Comment (YanZ777) [inline docs/adrs/0074-move-history-details-refactor.md:13]: **Suggestion:** This phrase doesn't make sense "to be able to three things:".

File: docs/adrs/0074-move-history-details-refactor.md
Comment (YanZ777) [reply docs/adrs/0074-move-history-details-refactor.md:15]: **suggestion:** The "the" can be removed.

File: docs/adrs/0074-move-history-details-refactor.md
Comment (YanZ777) [inline docs/adrs/0074-move-history-details-refactor.md:70]:
```suggestion
Existing event templates will not be affected by changes to Template Manager. For templates that cannot be matched using the default match settings, a new `getChangedValues` key can be added to the template object. This function that will return an array with the keys of the expected changed values. These changed values will be matched against the history record `changedValues`'s object keys in the template manager.
```

File: docs/adrs/0074-move-history-details-refactor.md
Comment (YanZ777) [inline docs/adrs/0074-move-history-details-refactor.md:99]:
```suggestion
* `-` *It assumes that no event will have the same action, event name, table name, and changed values. We may run into a case that meets this criteria and will have to come up with another metric in which to differentiate move events.*
```

File: docs/adrs/0074-move-history-details-refactor.md
Comment (YanZ777) [inline docs/adrs/0074-move-history-details-refactor.md:42]:
```suggestion
### 2. Match template to an event using the changed values.
```

File: docs/adrs/0074-move-history-details-refactor.md
Comment (YanZ777) [inline docs/adrs/0074-move-history-details-refactor.md:56]:
```suggestion
We will only match a template using `changedValues` if a move template includes a list of changed values. Otherwise, we will only use action, event and table name to match a template. With this check, existing move templates will not need to be updated to list their `changedValues`.
```

**Review by YanZ777 — COMMENTED** [review]
Thanks for writing up an ADR! 

I have a few grammar suggestions in the files itself.

With regards to the decision itself, how are we planning on handling events that have variable numbers of changed values, e.g. updateMTOShipment and the like?

**Comment by afoon** [comment]
> Thanks for writing up an ADR!
> 
> I have a few grammar suggestions in the files itself.
> 
> With regards to the decision itself, how are we planning on handling events that have variable numbers of changed values, e.g. updateMTOShipment and the like?

This is a very good question. I have to rethink my approach a bit so I'm moving this back into to a draft PR. Thank you for brining this to light 😊.

**Comment by afoon** [comment]
@YanZ777 @ronaktruss Thank you both for your questions in the first round of the review. I was able to step back and realize that there are two problems that arise when a move event matches to multiple templates. I've added an extended a background to further explain the issue. My first draft focuses on the first problem, but this second draft is more align with the second the problem.

File: docs/adrs/0074-move-history-details-refactor.md
Comment (michaelvaldes-truss) [inline docs/adrs/0074-move-history-details-refactor.md:64]: **suggestion**(non-blocking): you can probably remove these two lines since the following descriptions of the issues already have their own titles

File: docs/adrs/0074-move-history-details-refactor.md
Comment (michaelvaldes-truss) [inline docs/adrs/0074-move-history-details-refactor.md:89]: yaaaaaaaaaaas

**Review by donaldthai — APPROVED** [review]
🚀 🚢  Thanks for improving the codebase!

**Comment by afoon** [comment]
@ronaktruss and @YanZ777 , @donaldthai approved the ADR, but I will wait until the EOD to merge in case you all have any questions you want ask.

**Pull request merged**
